### PR TITLE
feat: add PermissionHub

### PR DESCRIPTION
**Pull request by cosinlink**
### Description

Currently, Greenfield supports creating buckets and groups from the smart contracts deployed on BSC/opBNB. This means that the owner of a bucket can be a contract address. However, contract addresses cannot upload objects or put policies under the bucket, resulting in empty buckets owned by smart contracts.

To address this issue, we are going to introduce the cross-chain permission module. With this BEP, smart contracts can grant EOA addresses the ability to upload objects under their buckets, and even support additional functions.

### Rationale

In the current framework, users could only change resource permission on Greenfield. The proposal allows Greenfield users to change resource permission directly on the BSC/opBNB network.

### Changes

Notable changes:
* PermissionHub Contract: A new middle-layer contract to request permission changes from BSC to Greenfield
* modify scripts for deploy and verify scan

### CODE REVIEW

File: contracts/interface/IPermissionHub.sol [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.0;
+
+import "../middle-layer/resource-mirror/storage/PermissionStorage.sol";
+
+interface IPermissionHub {
+    function deletePolicy(uint256 id) external payable returns (bool);
+
+    function createPutPolicy(
```
Comment (zjubfd) [inline contracts/interface/IPermissionHub.sol:10]: `createPutPolicy` ==> `putPolicy`

File: contracts/middle-layer/resource-mirror/PermissionHub.sol [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    }
+
+    /*----------------- initializer -----------------*/
+    function initialize(address _ERC721_token, address _additional) public initializer {
+        __cmn_hub_init_unchained(_ERC721_token, _additional);
+
+        channelId = PERMISSION_CHANNEL_ID;
+    }
+
+    function initializeV2() public reinitializer(2) {
```
Comment (zjubfd) [inline contracts/middle-layer/resource-mirror/PermissionHub.sol:22]: This is a newly deployed contract, no need to have `initializeV2`

File: contracts/interface/IPermissionHub.sol [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.0;
+
+import "../middle-layer/resource-mirror/storage/PermissionStorage.sol";
+
+interface IPermissionHub {
+    function deletePolicy(uint256 id) external payable returns (bool);
```
[the remark this answers is not in the document]
Comment (cosinlink) [reply contracts/interface/IPermissionHub.sol:8]: fixed

File: contracts/interface/IPermissionHub.sol (hunk 1 above)
Comment (cosinlink) [reply contracts/interface/IPermissionHub.sol:10]: fixed

File: contracts/middle-layer/resource-mirror/PermissionHub.sol [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+    function deletePolicy(uint256 id) external payable returns (bool) {
+        // check relay fee
+        (uint256 relayFee, uint256 minAckRelayFee) = ICrossChain(CROSS_CHAIN).getRelayFees();
+        require(msg.value >= relayFee + minAckRelayFee, "not enough fee");
+        uint256 _ackRelayFee = msg.value - relayFee;
+
+        // check authorization
+        address owner = IERC721NonTransferable(ERC721Token).ownerOf(id);
+        require(
+            msg.sender == owner ||
```
[the remark this answers is not in the document]
Comment (cosinlink) [reply contracts/middle-layer/resource-mirror/PermissionHub.sol]: fixed

File: contracts/middle-layer/resource-mirror/PermissionHub.sol (hunk 2 above)
Comment (cosinlink) [reply contracts/middle-layer/resource-mirror/PermissionHub.sol:22]: In order to keep the code completely consistent with the BucketHub, GroupHub and ObjectHub, this makes it easier to compare the code difference

File: contracts/Config.sol [hunk 5]
```diff
@@ -11,20 +11,22 @@ abstract contract Config {
...
-    address public constant CROSS_CHAIN = address(0);
-    address public constant TOKEN_HUB = address(0);
-    address public constant LIGHT_CLIENT = address(0);
-    address public constant RELAYER_HUB = address(0);
-    address public constant BUCKET_HUB = address(0);
-    address public constant OBJECT_HUB = address(0);
-    address public constant GROUP_HUB = address(0);
-    address public constant EMERGENCY_OPERATOR = address(0);
-    address public constant EMERGENCY_UPGRADE_OPERATOR = address(0);
+    address public constant PROXY_ADMIN = 0x212fdfCfCC22db97DeB3AC3260414909282BB4EE;
```
Comment (zjubfd) [inline contracts/Config.sol:18]: Please revert this.

**Review by Pythonberg1997 — APPROVED** [review]

**Review by zjubfd — APPROVED** [review]

**Pull request merged**